### ISSUE TITLE
When seeing dataset operations (airflow capture_executions enabled. Reads/Writes tab available), an Unknown error occurred (code 500).

### ISSUE THREAD

**Issue by sb-sebkim** [datahub-project/datahub#5191]
**Describe the bug**
When seeing dataset operations (airflow capture_executions enabled. Reads/Writes tab available), an Unknown error occurred (code 500). Recent DataHub introduces capture_executions which enables showing dataset Read/Writes operations in Dataset views and also airflow task run histories in DataJob views. This is normally working when the amount of lineage metadata is light, however when this amount is heavy (a lot of upstreams) it throws an 'Unknown error occurred (code 500)'.
<img width="1792" alt="스크린샷 2022-06-17 오전 9 27 38" src="https://user-images.githubusercontent.com/75765303/174199244-53d31734-7c0a-43bc-9cf2-62a73df01da0.png">

**To Reproduce**

1. Set the capture_executions to true for the airflow side like following:
config:
  lineage:
    backend: 'datahub_provider.lineage.datahub.DatahubLineageBackend'
    datahub_kwargs: '{{ printf "{\"datahub_conn_id\": \"datahub_rest_default\",
    \"cluster\": \"%s\",
    \"capture_ownership_info\": true,
    \"capture_tags_info\": true,
    \"capture_executions\": true,
    \"graceful_exceptions\": true }" (upper .Values.environ) }}'

2. Ingest an airflow task (DataJob) that has a lot of upstreams. For example, `DataJob A` has 80 upstreams called `Dataset U{x}` (e.g. U1, U2, ... U80) and one downstream called `Dataset D`.

3. Visit the `Dataset D` in DataHub web UI, and Click `Operations`. Because `Dataset D` is the downstream of the `DataJob A`, there would be empty `Reads`. Click `Writes` tab and the error is shown. 
 

**Expected behavior**
`Operations` tab works as expected.

**Screenshots**

**Desktop (please complete the following information):**
 - OS: macOS
 - Browser: Chrome
 - Version: 101.0.4951.54

**Additional context**
When clicking `Writes` operations in the Dataset view, it tries to fetch graphql query like following.

```
query getDatasetRuns($urn: String!, $start: Int!, $count: Int!, $direction: RelationshipDirection!) {
  dataset(urn: $urn) {
    runs(start: $start, count: $count, direction: $direction) {
      ...runResults
      __typename
    }
    __typename
  }
}

fragment runResults on DataProcessInstanceResult {
  count
  start
  total
  runs {
    urn
    type
    created {
      time
      actor
      __typename
    }
    name
    state(startTimeMillis: null, endTimeMillis: null, limit: 1) {
      status
      attempt
      result {
        resultType
        nativeResultType
        __typename
      }
      timestampMillis
      __typename
    }
    inputs: relationships(
      input: {types: ["Consumes"], direction: OUTGOING, start: 0, count: 20}
    ) {
      ...fullRelationshipResults
      __typename
    }
    outputs: relationships(
      input: {types: ["Produces"], direction: OUTGOING, start: 0, count: 20}
    ) {
      ...fullRelationshipResults
      __typename
    }
    parentTemplate: relationships(
      input: {types: ["InstanceOf"], direction: OUTGOING, start: 0, count: 1}
    ) {
      ...fullRelationshipResults
      __typename
    }
    externalUrl
    __typename
  }
  __typename
}

fragment fullRelationshipResults on EntityRelationshipsResult {
  start
  count
  total
  relationships {
    type
    direction
    entity {
      ...relationshipFields
      __typename
    }
    __typename
  }
  __typename
}

fragment relationshipFields on EntityWithRelationships {
  urn
  type
  ... on DataJob {
    ...dataJobFields
    editableProperties {
      description
      __typename
    }
    status {
      removed
      __typename
    }
    __typename
  }
  ... on DataFlow {
    orchestrator
    flowId
    cluster
    properties {
      name
      description
      project
      __typename
    }
    ownership {
      ...ownershipFields
      __typename
    }
    globalTags {
      ...globalTagsFields
      __typename
    }
    glossaryTerms {
      ...glossaryTerms
      __typename
    }
    editableProperties {
      description
      __typename
    }
    platform {
      ...platformFields
      __typename
    }
    domain {
      ...entityDomain
      __typename
    }
    status {
      removed
      __typename
    }
    __typename
  }
  ... on Dashboard {
    ...dashboardFields
    editableProperties {
      description
      __typename
    }
    platform {
      ...platformFields
      __typename
    }
    status {
      removed
      __typename
    }
    __typename
  }
  ... on Chart {
    tool
    chartId
    properties {
      name
      description
      __typename
    }
    editableProperties {
      description
      __typename
    }
    ownership {
      ...ownershipFields
      __typename
    }
    platform {
      ...platformFields
      __typename
    }
    domain {
      ...entityDomain
      __typename
    }
    status {
      removed
      __typename
    }
    __typename
  }
  ... on Dataset {
    name
    properties {
      name
      description
      __typename
    }
    editableProperties {
      description
      __typename
    }
    platform {
      ...platformFields
      __typename
    }
    ownership {
      ...ownershipFields
      __typename
    }
    subTypes {
      typeNames
      __typename
    }
    status {
      removed
      __typename
    }
    __typename
  }
  ... on MLModelGroup {
    urn
    type
    name
    description
    origin
    platform {
      ...platformFields
      __typename
    }
    ownership {
      ...ownershipFields
      __typename
    }
    status {
      removed
      __typename
    }
    __typename
  }
  ... on MLModel {
    urn
    type
    name
    description
    origin
    platform {
      ...platformFields
      __typename
    }
    ownership {
      ...ownershipFields
      __typename
    }
    status {
      removed
      __typename
    }
    __typename
  }
  ... on MLFeatureTable {
    ...nonRecursiveMLFeatureTable
    __typename
  }
  ... on MLFeature {
    ...nonRecursiveMLFeature
    __typename
  }
  ... on MLPrimaryKey {
    ...nonRecursiveMLPrimaryKey
    __typename
  }
  upstream: lineage(input: {direction: UPSTREAM, start: 0, count: 100}) {
    ...leafLineageResults
    __typename
  }
  downstream: lineage(input: {direction: DOWNSTREAM, start: 0, count: 100}) {
    ...leafLineageResults
    __typename
  }
  __typename
}

fragment dataJobFields on DataJob {
  urn
  type
  dataFlow {
    ...nonRecursiveDataFlowFields
    __typename
  }
  jobId
  ownership {
    ...ownershipFields
    __typename
  }
  inputOutput {
    inputDatasets {
      ...nonRecursiveDatasetFields
      __typename
    }
    outputDatasets {
      ...nonRecursiveDatasetFields
      __typename
    }
    inputDatajobs {
      ...nonRecursiveDataJobFields
      __typename
    }
    __typename
  }
  properties {
    name
    description
    externalUrl
    customProperties {
      key
      value
      __typename
    }
    __typename
  }
  editableProperties {
    description
    __typename
  }
  globalTags {
    ...globalTagsFields
    __typename
  }
  institutionalMemory {
    ...institutionalMemoryFields
    __typename
  }
  glossaryTerms {
    ...glossaryTerms
    __typename
  }
  domain {
    ...entityDomain
    __typename
  }
  deprecation {
    ...deprecationFields
    __typename
  }
  dataPlatformInstance {
    ...dataPlatformInstanceFields
    __typename
  }
  __typename
}

fragment nonRecursiveDataFlowFields on DataFlow {
  urn
  type
  orchestrator
  flowId
  cluster
  properties {
    name
    description
    project
    externalUrl
    customProperties {
      key
      value
      __typename
    }
    __typename
  }
  editableProperties {
    description
    __typename
  }
  ownership {
    ...ownershipFields
    __typename
  }
  platform {
    ...platformFields
    __typename
  }
  domain {
    ...entityDomain
    __typename
  }
  deprecation {
    ...deprecationFields
    __typename
  }
  __typename
}

fragment ownershipFields on Ownership {
  owners {
    owner {
      ... on CorpUser {
        urn
        type
        username
        info {
          active
          displayName
          title
          email
          firstName
          lastName
          fullName
          __typename
        }
        properties {
          active
          displayName
          title
          email
          firstName
          lastName
          fullName
          __typename
        }
        editableProperties {
          displayName
          title
          pictureLink
          email
          __typename
        }
        __typename
      }
      ... on CorpGroup {
        urn
        type
        name
        properties {
          displayName
          email
          __typename
        }
        info {
          displayName
          email
          admins {
            urn
            username
            info {
              active
              displayName
              title
              email
              firstName
              lastName
              fullName
              __typename
            }
            editableInfo {
              pictureLink
              teams
              skills
              __typename
            }
            __typename
          }
          members {
            urn
            username
            info {
              active
              displayName
              title
              email
              firstName
              lastName
              fullName
              __typename
            }
            editableInfo {
              pictureLink
              teams
              skills
              __typename
            }
            __typename
          }
          groups
          __typename
        }
        __typename
      }
      __typename
    }
    type
    __typename
  }
  lastModified {
    time
    __typename
  }
  __typename
}

fragment platformFields on DataPlatform {
  urn
  type
  name
  properties {
    type
    displayName
    datasetNameDelimiter
    logoUrl
    __typename
  }
  displayName
  info {
    type
    displayName
    datasetNameDelimiter
    logoUrl
    __typename
  }
  __typename
}

fragment entityDomain on Domain {
  urn
  properties {
    name
    __typename
  }
  __typename
}

fragment deprecationFields on Deprecation {
  actor
  deprecated
  note
  decommissionTime
  __typename
}

fragment nonRecursiveDatasetFields on Dataset {
  urn
  name
  type
  origin
  uri
  platform {
    ...platformFields
    __typename
  }
  dataPlatformInstance {
    ...dataPlatformInstanceFields
    __typename
  }
  platformNativeType
  properties {
    name
    description
    customProperties {
      key
      value
      __typename
    }
    externalUrl
    __typename
  }
  editableProperties {
    description
    __typename
  }
  ownership {
    ...ownershipFields
    __typename
  }
  institutionalMemory {
    ...institutionalMemoryFields
    __typename
  }
  schemaMetadata(version: 0) {
    ...schemaMetadataFields
    __typename
  }
  previousSchemaMetadata: schemaMetadata(version: -1) {
    ...schemaMetadataFields
    __typename
  }
  editableSchemaMetadata {
    editableSchemaFieldInfo {
      fieldPath
      description
      globalTags {
        ...globalTagsFields
        __typename
      }
      __typename
    }
    __typename
  }
  globalTags {
    ...globalTagsFields
    __typename
  }
  glossaryTerms {
    ...glossaryTerms
    __typename
  }
  subTypes {
    typeNames
    __typename
  }
  domain {
    ...entityDomain
    __typename
  }
  container {
    ...entityContainer
    __typename
  }
  deprecation {
    ...deprecationFields
    __typename
  }
  __typename
}

fragment dataPlatformInstanceFields on DataPlatformInstance {
  urn
  type
  platform {
    ...platformFields
    __typename
  }
  instanceId
  __typename
}

fragment institutionalMemoryFields on InstitutionalMemory {
  elements {
    url
    author {
      urn
      username
      __typename
    }
    description
    created {
      actor
      time
      __typename
    }
    __typename
  }
  __typename
}

fragment schemaMetadataFields on SchemaMetadata {
  aspectVersion
  createdAt
  datasetUrn
  name
  platformUrn
  version
  cluster
  hash
  platformSchema {
    ... on TableSchema {
      schema
      __typename
    }
    ... on KeyValueSchema {
      keySchema
      valueSchema
      __typename
    }
    __typename
  }
  fields {
    fieldPath
    jsonPath
    nullable
    description
    type
    nativeDataType
    recursive
    isPartOfKey
    globalTags {
      ...globalTagsFields
      __typename
    }
    glossaryTerms {
      ...glossaryTerms
      __typename
    }
    __typename
  }
  primaryKeys
  foreignKeys {
    name
    sourceFields {
      fieldPath
      __typename
    }
    foreignFields {
      fieldPath
      __typename
    }
    foreignDataset {
      urn
      name
      type
      origin
      uri
      properties {
        description
        __typename
      }
      platform {
        ...platformFields
        __typename
      }
      platformNativeType
      ownership {
        ...ownershipFields
        __typename
      }
      globalTags {
        ...globalTagsFields
        __typename
      }
      glossaryTerms {
        ...glossaryTerms
        __typename
      }
      __typename
    }
    __typename
  }
  __typename
}

fragment globalTagsFields on GlobalTags {
  tags {
    tag {
      urn
      name
      description
      properties {
        colorHex
        __typename
      }
      __typename
    }
    __typename
  }
  __typename
}

fragment glossaryTerms on GlossaryTerms {
  terms {
    term {
      ...glossaryTerm
      __typename
    }
    __typename
  }
  __typename
}

fragment glossaryTerm on GlossaryTerm {
  urn
  name
  type
  hierarchicalName
  properties {
    name
    definition
    termSource
    __typename
  }
  __typename
}

fragment entityContainer on Container {
  urn
  platform {
    ...platformFields
    __typename
  }
  properties {
    name
    __typename
  }
  subTypes {
    typeNames
    __typename
  }
  deprecation {
    ...deprecationFields
    __typename
  }
  __typename
}

fragment nonRecursiveDataJobFields on DataJob {
  urn
  properties {
    name
    description
    externalUrl
    customProperties {
      key
      value
      __typename
    }
    __typename
  }
  globalTags {
    ...globalTagsFields
    __typename
  }
  domain {
    ...entityDomain
    __typename
  }
  deprecation {
    ...deprecationFields
    __typename
  }
  __typename
}

fragment dashboardFields on Dashboard {
  urn
  type
  tool
  dashboardId
  properties {
    name
    description
    customProperties {
      key
      value
      __typename
    }
    externalUrl
    access
    lastRefreshed
    created {
      time
      __typename
    }
    lastModified {
      time
      __typename
    }
    __typename
  }
  ownership {
    ...ownershipFields
    __typename
  }
  globalTags {
    ...globalTagsFields
    __typename
  }
  institutionalMemory {
    ...institutionalMemoryFields
    __typename
  }
  glossaryTerms {
    ...glossaryTerms
    __typename
  }
  platform {
    ...platformFields
    __typename
  }
  domain {
    ...entityDomain
    __typename
  }
  container {
    ...entityContainer
    __typename
  }
  parentContainers {
    ...parentContainersFields
    __typename
  }
  status {
    removed
    __typename
  }
  deprecation {
    ...deprecationFields
    __typename
  }
  dataPlatformInstance {
    ...dataPlatformInstanceFields
    __typename
  }
  __typename
}

fragment parentContainersFields on ParentContainersResult {
  count
  containers {
    ...entityContainer
    __typename
  }
  __typename
}

fragment nonRecursiveMLFeatureTable on MLFeatureTable {
  urn
  type
  name
  platform {
    ...platformFields
    __typename
  }
  dataPlatformInstance {
    ...dataPlatformInstanceFields
    __typename
  }
  description
  properties {
    description
    mlFeatures {
      ...nonRecursiveMLFeature
      __typename
    }
    mlPrimaryKeys {
      ...nonRecursiveMLPrimaryKey
      __typename
    }
    customProperties {
      key
      value
      __typename
    }
    __typename
  }
  ownership {
    ...ownershipFields
    __typename
  }
  institutionalMemory {
    ...institutionalMemoryFields
    __typename
  }
  status {
    removed
    __typename
  }
  glossaryTerms {
    ...glossaryTerms
    __typename
  }
  domain {
    ...entityDomain
    __typename
  }
  tags {
    ...globalTagsFields
    __typename
  }
  editableProperties {
    description
    __typename
  }
  deprecation {
    ...deprecationFields
    __typename
  }
  __typename
}

fragment nonRecursiveMLFeature on MLFeature {
  urn
  type
  name
  featureNamespace
  description
  dataType
  properties {
    description
    dataType
    version {
      versionTag
      __typename
    }
    sources {
      urn
      name
      type
      origin
      description
      uri
      platform {
        ...platformFields
        __typename
      }
      platformNativeType
      __typename
    }
    __typename
  }
  ownership {
    ...ownershipFields
    __typename
  }
  institutionalMemory {
    ...institutionalMemoryFields
    __typename
  }
  status {
    removed
    __typename
  }
  glossaryTerms {
    ...glossaryTerms
    __typename
  }
  domain {
    ...entityDomain
    __typename
  }
  tags {
    ...globalTagsFields
    __typename
  }
  editableProperties {
    description
    __typename
  }
  deprecation {
    ...deprecationFields
    __typename
  }
  dataPlatformInstance {
    ...dataPlatformInstanceFields
    __typename
  }
  featureTables: relationships(
    input: {types: ["Contains"], direction: INCOMING, start: 0, count: 100}
  ) {
    relationships {
      type
      entity {
        ... on MLFeatureTable {
          platform {
            ...platformFields
            __typename
          }
          __typename
        }
        __typename
      }
      __typename
    }
    __typename
  }
  __typename
}

fragment nonRecursiveMLPrimaryKey on MLPrimaryKey {
  urn
  type
  name
  featureNamespace
  description
  dataType
  properties {
    description
    dataType
    version {
      versionTag
      __typename
    }
    sources {
      urn
      name
      type
      origin
      description
      uri
      platform {
        ...platformFields
        __typename
      }
      platformNativeType
      __typename
    }
    __typename
  }
  ownership {
    ...ownershipFields
    __typename
  }
  institutionalMemory {
    ...institutionalMemoryFields
    __typename
  }
  status {
    removed
    __typename
  }
  glossaryTerms {
    ...glossaryTerms
    __typename
  }
  domain {
    ...entityDomain
    __typename
  }
  tags {
    ...globalTagsFields
    __typename
  }
  editableProperties {
    description
    __typename
  }
  deprecation {
    ...deprecationFields
    __typename
  }
  dataPlatformInstance {
    ...dataPlatformInstanceFields
    __typename
  }
  featureTables: relationships(
    input: {types: ["KeyedBy"], direction: INCOMING, start: 0, count: 100}
  ) {
    relationships {
      type
      entity {
        ... on MLFeatureTable {
          platform {
            ...platformFields
            __typename
          }
          __typename
        }
        __typename
      }
      __typename
    }
    __typename
  }
  __typename
}

fragment leafLineageResults on EntityLineageResult {
  start
  count
  total
  relationships {
    type
    entity {
      urn
      type
      __typename
    }
    __typename
  }
  __typename
}

```

with variable:
```
{
  "count": 20,
  "direction": "OUTGOING",
  "start": 0,
  "urn": "urn:li:dataset:(urn:li:dataPlatform:bigquery,sb-dw-dp-prod.processed_core_metrics.app_daily_feature_usage,PROD)"
}
```

If we decrease the count from 20 to 10, then it works!

**Comment by gabe-lyons** [comment]
Addressed by https://github.com/datahub-project/datahub/pull/5632

**Comment by gabe-lyons** [comment]
Resolved by #5632!
